Mark optional steps in Host Registration (#2210)
theforeman/foreman-documentation: guides/common/modules/proc_registering-hosts.adoc

# curl -sS --insecure https://{foreman-example-com}/register ...
----
. Select the *Advanced* tab.

> . Optional: From the *Setup REX* list, select whether you want to deploy {Project} SSH keys to hosts or not.

Reviewer: ```suggestion
. If you want to deploy {Project} SSH keys to your hosts, select the *Setup REX* checkbox.
```

I don't believe this step is optional. I'd rather say it's conditional depening on your use case.

IMHO only descriptions etc. are truly optional because they don't really affect the outcome.

Author: By optional it's meant that the curl command can be generated regardless of filling in this field. Which it can.

Reviewer: But that's an active descision to say "I don't want XY". Does IBM Style Guide say anything about that?

Author: It isn't the same case as with the SmartProxy prerequisite.
"Optional" says: Foreman doesn't require the value for this function to work, it's up to you if you set it or not, and it will affect the function's behavior in this or that way. If the user will leave it as it is, the function (registration in this case) will still work.
In case of the prerequisite, the user had to fulfill it in a certain scenario, otherwise the following procedure wouldn't work.
This is why I would stick with my original proposal.

Author: REX setup is an addition to registration. Regardless of the REX setup, the user will be able to register the host, that's why it should be optional.
The whole registration needs rewriting, I would keep it simple for now and get to rewriting it later.
